source/slang/glsl.meta.slang

```diff
@@ -417,3 +252,2047 @@ ${{{{
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public int textureSize(sampler1DShadow sampler, int lod)
+{
+    int result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result, numberOfLevels);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(sampler2DShadow sampler, int lod)
+{
+    vector<int,2> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, numberOfLevels);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(samplerCubeShadow sampler, int lod)
+{
+    vector<int,2> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, numberOfLevels);
+    return result;
+}
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec3 textureSize(SamplerCubeArray<vector<T,N>> sampler, int lod)
+{
+    vector<int,3> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, result.z, numberOfLevels);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec3 textureSize(samplerCubeArrayShadow sampler, int lod)
+{
+    vector<int,3> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, result.z, numberOfLevels);
+    return result;
+}
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(Sampler2DRect<vector<T,N>> sampler)
+{
+    vector<int,2> result;
+    sampler.GetDimensions(result.x, result.y);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(sampler2DRectShadow sampler)
+{
+    vector<int,2> result;
+    int numberOfLevels;
+    sampler.GetDimensions(result.x, result.y);
+    return result;
+}
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(Sampler1DArray<vector<T,N>> sampler, int lod)
+{
+    vector<int,2> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, numberOfLevels);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec2 textureSize(sampler1DArrayShadow sampler, int lod)
+{
+    vector<int,2> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, numberOfLevels);
+    return result;
+}
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec3 textureSize(Sampler2DArray<vector<T,N>> sampler, int lod)
+{
+    vector<int,3> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, result.z, numberOfLevels);
+    return result;
+}
+
+__target_intrinsic(glsl)
+[ForceInline]
+public ivec3 textureSize(sampler2DArrayShadow sampler, int lod)
+{
+    vector<int,3> result;
+    int numberOfLevels;
+    sampler.GetDimensions(lod, result.x, result.y, result.z, numberOfLevels);
+    return result;
+}
+
+#if defined(TEST_samplerBuffer)
```

Comment:
Is `samplerBuffer` a thing in glsl?

Reply:
Yes samplerBuffer is defined in OpenGL 4.6 spec. That is the Shadet Storage Buffer Object.